documents/data-sharing.md

```diff
@@ -0,0 +1,58 @@
+# Annual survey data sharing
+
+This document describes how data from the annual survey should be shared.
+
+## Objectives
+
+In priority order:
+
+* Protect the privacy of respondents.
+* Provide high quality data (unbiased data which is unlikely to be misinterpreted, c.f., raw data).
+* Provide useful and actionable data to the Rust project (Core team, Foundation, teams and working groups, contributors).
+* Provide useful data for users and potential users of Rust and providers of Rust tools and services.
```

Comment:
I might try to reformulate this as seeking to provide summaries to this group -- I am not sure it should be a goal to share un-digested 'raw' data (even if it's in e.g. "this question had X% responding with Y column" aggregate, particularly with fully external parties to the project.

Mostly this is to avoid a situation where we accidentally over-share, so maybe this is just 'low enough' that priority 1 (privacy) is sufficient, but I might still reformulate. I'm not sure we should be spending time as a project on assembling data for fully public consumption.

Reply:
I am trying to list the objectives of sharing in a general sense, so here I mean 'data' in an abstract way, rather than specifically the survey data.

> Mostly this is to avoid a situation where we accidentally over-share, so maybe this is just 'low enough' that priority 1 (privacy) is sufficient, but I might still reformulate. 

Yeah, these objectives are in priority order, so we shouldn't try to achieve any of the others at the expense of privacy.

> I'm not sure we should be spending time as a project on assembling data for fully public consumption.

I disagree with this though - if it benefits the project, by helping people adopt Rust or helps persuade companies to produce tools for Rust, etc., then that seems like a good use of our time.